src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private int defaultSize = 10;
+    private int maxSize = 50;
+    private int maxBound = 5;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    int index = random.nextInt(maxBound);
```

Comment:
The variable `index` should be local to the `getRandomFigure` method since it's only used there. Also, generating the index should happen within the `getRandomFigure` method to ensure a different figure is provided each time the method is called.